src/libstore/unix/build/hook-instance.cc

```diff
@@ -16,11 +16,23 @@ HookInstance::HookInstance()
     if (buildHookArgs.empty())
         throw Error("'build-hook' setting is empty");
 
-    auto buildHook = canonPath(buildHookArgs.front());
+    std::filesystem::path buildHook = buildHookArgs.front();
     buildHookArgs.pop_front();
 
+    if (buildHook.is_absolute()) {
+        buildHook = std::filesystem::weakly_canonical(buildHook);
+    } else {
+        for (auto & dir : tokenizeString<Strings>(getEnv("PATH").value_or(""), ":")) {
```

Comment:
```suggestion
        for (auto & dir : splitString<Strings>(getEnv("PATH").value_or(""), ":")) {
```

I have a tested version of `PATH` resolution waiting to become a PR after https://github.com/NixOS/nix/pull/11093 and before fixing #11021, so no rush I guess?